go/internal/services/ratelimit/service.go

```diff
@@ -330,21 +332,13 @@ func (s *service) Ratelimit(ctx context.Context, req RatelimitRequest) (Ratelimi
 		}, nil
 	}
 
-	// If we get here, the request is allowed
-	// Increment current window counter
-	currentWindow.counter += req.Cost
```

Comment:
we no longer decrement the counter, which does lead for over-allowing requests if we don't sync from redis in the meantime.

Also our rollback is pointless since we never decremented the counter to begin with.

This only works because our replay buffering sets the window counter aswell if it notices that it doesn't match